### PR TITLE
feat(LUKE-46): integrate existing Claude Code sessions

### PR DESCRIPTION
**Pull request by dastratakos**
## Summary

- Add a passive Claude Code main-process adapter that discovers local Claude session files without hooks, wrappers, MCP, controls, or provider-owned writes.
- Feed bounded, redacted observations into the shared session registry and expose normalized session state through narrow IPC.
- Show live sessions in the notch panel, with smoke fixtures limited to capture/evidence mode.
- Add synthetic adapter tests for transcript non-leakage, stale states, duplicate ids, and missing Claude data.

## Evidence

- Platform-independent checks: `./scripts/check.sh` passed
- macOS Electron verification (`./scripts/verify.sh`): embedded portable checks passed, then stopped with `error: this command requires macOS` in this Linux environment

### Automated visual evidence

[Download the deterministic macOS evidence](https://github.com/ReviewStage/luke/actions/runs/31550579789/artifacts/9124101240) · [workflow run](https://github.com/ReviewStage/luke/actions/runs/31550579789)

- Commit: `bfdf16c8b662e5f8b6b877e21985946043e24c1d`
- Scenario: `smoke`
- Physical-notch check: not performed by CI

### Physical-device evidence

- Screenshot or screen recording: `not attached`
- Physical-notch check: `not performed`
- Device/display configuration: `not recorded`

## Notes

- Local macOS smoke testing was reported working by the requester before this PR was opened.
- Blockers or follow-up verification: macOS visual evidence must come from CI or a physical Mac.

---

[Open workspace in Conductor](https://app.conductor.build/workspace/4b6e5029-fb0d-404c-a6d2-33bacf391a09)

### CODE REVIEW

File: apps/desktop/src/renderer/index.tsx [hunk 1]
```diff
@@ -122,8 +143,44 @@ function notchStyle(display: DisplayDiagnostic): CSSProperties {
...
+  return (
+    session.status === SESSION_STATUS.WAITING ||
+    session.attention.disposition !== ATTENTION_DISPOSITION.SILENT
+  );
+}
+
+function sessionState(session: NormalizedSession): SessionState {
+  if (sessionNeedsAttention(session)) return "attention";
+  if (session.status === SESSION_STATUS.COMPLETE) return "complete";
+  return "working";
```
Comment (dastratakos) [inline apps/desktop/src/renderer/index.tsx]: should we be using enums instead of returning strings here?

File: apps/desktop/src/claude-code-adapter.ts [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+  now: number,
+  activeSessionFreshnessMs: number,
+): ProviderSessionObservation["status"] {
+  if (parsed.eventType === CLAUDE_EVENT_TYPE.RESULT) return SESSION_STATUS.COMPLETE;
+  if (parsed.eventType === CLAUDE_EVENT_TYPE.ASSISTANT) return SESSION_STATUS.WAITING;
+  if (parsed.eventType === CLAUDE_EVENT_TYPE.USER) {
+    return now - observedAt <= activeSessionFreshnessMs
+      ? SESSION_STATUS.WORKING
+      : SESSION_STATUS.UNKNOWN;
+  }
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply apps/desktop/src/claude-code-adapter.ts:271]: Fixed in eaffc2c: assistant-tail observations now use the same freshness window as user-tail observations, so stale Claude replies become unknown instead of staying in WAITING/attention. Added a regression test for stale assistant tails.

File: apps/desktop/src/renderer/index.tsx (hunk 1 above)
Comment (dastratakos) [reply apps/desktop/src/renderer/index.tsx]: Addressed in eaffc2c by adding a shared SESSION_STATE value set in @sidecar/core and using those constants in the renderer instead of returning raw state strings.

File: apps/desktop/src/claude-code-adapter.ts [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+  const isFresh = now - observedAt <= activeSessionFreshnessMs;
+  if (parsed.eventType === CLAUDE_EVENT_TYPE.RESULT) return SESSION_STATUS.COMPLETE;
+  if (parsed.eventType === CLAUDE_EVENT_TYPE.ASSISTANT) {
+    return isFresh ? SESSION_STATUS.WAITING : SESSION_STATUS.UNKNOWN;
+  }
+  if (parsed.eventType === CLAUDE_EVENT_TYPE.USER) {
+    return isFresh ? SESSION_STATUS.WORKING : SESSION_STATUS.UNKNOWN;
+  }
+  return SESSION_STATUS.UNKNOWN;
+}
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply apps/desktop/src/claude-code-adapter.ts:273]: Fixed in e7ce515: status parsing now ignores non-status Claude records such as system/summary, preserving the last user/assistant/result event. Added a system-tail regression test.

File: apps/desktop/src/renderer/index.tsx [hunk 4]
```diff
@@ -122,8 +144,44 @@ function notchStyle(display: DisplayDiagnostic): CSSProperties {
...
+  return (
+    session.status === SESSION_STATUS.WAITING ||
+    session.attention.disposition !== ATTENTION_DISPOSITION.SILENT
+  );
+}
+
+function sessionState(session: NormalizedSession): SessionState {
+  if (sessionNeedsAttention(session)) return SESSION_STATE.ATTENTION;
+  if (session.status === SESSION_STATUS.COMPLETE) return SESSION_STATE.COMPLETE;
+  return SESSION_STATE.WORKING;
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply apps/desktop/src/renderer/index.tsx:159]: Fixed in e7ce515: SESSION_STATUS.UNKNOWN now maps to a neutral SESSION_STATE.UNKNOWN display state, with neutral indicator styling and copy that says no active sessions are observed.

File: apps/desktop/src/claude-code-adapter.ts [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply apps/desktop/src/claude-code-adapter.ts:273]: Fixed in bfdf16c: fresh assistant records containing tool_use content are now classified as WORKING rather than WAITING, with test coverage for assistant tool-use messages.

File: apps/desktop/src/claude-code-adapter.ts [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+  } finally {
+    await handle?.close();
+  }
+}
+
+function tailLines(tail: string): string[] {
+  const lines = tail.split(/\r?\n/).filter((line) => line.trim().length > 0);
+  if (!tail.startsWith("{")) lines.shift();
+  return lines;
+}
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply apps/desktop/src/claude-code-adapter.ts:192]: Fixed in bfdf16c: if the bounded tail has no complete status-bearing event but the session file is fresh, observation reports WORKING instead of UNKNOWN. Added a regression test that starts the tail inside a large tool result.
